test/components/views/beacon/BeaconMarker-test.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { mount } from 'enzyme';
+import maplibregl from 'maplibre-gl';
+import { act } from 'react-dom/test-utils';
+import {
+    Beacon,
+    Room,
+    RoomMember,
+    getBeaconInfoIdentifier,
+} from 'matrix-js-sdk/src/matrix';
+
+import BeaconMarker from '../../../../src/components/views/beacon/BeaconMarker';
+import MatrixClientContext from '../../../../src/contexts/MatrixClientContext';
+import { getMockClientWithEventEmitter, makeBeaconEvent, makeBeaconInfoEvent } from '../../../test-utils';
+
+describe('<BeaconMarker />', () => {
+    // 14.03.2022 16:15
+    const now = 1647270879403;
+    // stable date for snapshots
+    jest.spyOn(global.Date, 'now').mockReturnValue(now);
+    const roomId = '!room:server';
+    const aliceId = '@alice:server';
+
+    const aliceMember = new RoomMember(roomId, aliceId);
+
+    const mockMap = new maplibregl.Map();
+
+    const mockClient = getMockClientWithEventEmitter({
+        getClientWellKnown: jest.fn().mockReturnValue({
+            "m.tile_server": { map_style_url: 'maps.com' },
```

Comment:
I've mentioned this elsewhere, and apologies for only noticing it now, but `m.tile_server` is an illegal value in the spec so this needs to be renamed. A different PR can take care of that.

Reply:
oh, good catch! opened a PR to rename https://github.com/matrix-org/matrix-react-sdk/pull/8316